### DISCUSSION TITLE
RPATH modification rationale?

### DISCUSSION THREAD

**Discussion by moradology** [lambgeo/docker-lambda#68] [Q&A] [answered]
The work in this repo is really solid and I'm fairly certain it will serve as a useful reference for the geospatial community. With that said, I'm curious about the rpath modifications done in the main dockerfile here:
https://github.com/lambgeo/docker-lambda/blob/master/dockerfiles/Dockerfile#L386-L401

What's this part doing? GDAL seems to function nicely without it, but perhaps I'm not fully exercising relevant functions that will be upset when they can't find headers at runtime.

For context, it appears as though at least some of the custom EMR images I'm aiming to build gdal against have *completely* broken perl installations which cause `patchelf` builds to fall over. Like: necessary pragma files (e.g. strict.pm) just aren't there.

**Comment by vincentsarago** [comment, marked as answer]
> What's this part doing? GDAL seems to function nicely without it, but perhaps I'm not fully exercising relevant functions that will be upset when they can't find headers at runtime.

the `rpath` modification is done to make sure we keep a relative reference for the libs when we package lambda layer or package 

If the goal is to build a pure docker image the rpath usage can be removed

**Reply by moradology** [reply]
Somehow didn't even realize issues/discussions were now a separate thing. Thanks for the (incredibly speedy) reply
